Backend/Tumultu/Architecture.Tests/ArchitectureTests.cs

```diff
@@ -0,0 +1,231 @@
+using MediatR;
+using NetArchTest.Rules;
+using System.Reflection;
+
+namespace Architecture.Tests;
+
+public class ArchitectureTests
+{
+    private const string Domain = "Tumultu.Domain";
+    private const string Application = "Tumultu.Application";
+    private const string Infrastructure = "Tumultu.Infrastructure";
+    private const string Presentation = "Tumultu.Presentation";
+
+    private readonly Dictionary<string, Assembly> _assemblies = new Dictionary<string, Assembly>()
+    {
+        { Domain, typeof(Tumultu.Domain.AssemblyReference).Assembly },
+        { Application, typeof(Tumultu.Application.AssemblyReference).Assembly },
+        { Infrastructure, typeof(Tumultu.Infrastructure.AssemblyReference).Assembly },
+        { Presentation, typeof(Tumultu.Presentation.AssemblyReference).Assembly }
+    };
+
+    [Fact]
+    public void Domain_Should_NotHaveDependencyOnOtherProjects()
+    {
+        // Arrange
+        var domainAssembly = _assemblies[Domain];
+
+        // Act
+        TestResult result = Types
+            .InAssembly(domainAssembly)
+            .Should()
+            .NotHaveDependencyOnAll(Application, Infrastructure, Presentation)
+            .GetResult();
+
+        // Assert
+        result.IsSuccessful.Should().BeTrue();
+    }
+
+    [Fact]
+    public void Application_Should_NotHaveDependencyOnOtherProjects()
```

Comment:
The name is misleading since Application is supposed to have dependency on Domain